src/linux/sensor_ds18b20.c

```diff
@@ -217,10 +224,15 @@ ds18_send_and_request(struct ds18_s *d, uint32_t next_begin_time, uint8_t oid)
     if (d->status == W1_ERROR) {
         // try_shutdown expects a static string. Output the specific error,
         // then shut down with a generic error.
-        output("Error: %s", d->error);
         pthread_mutex_unlock(&d->lock);
-        try_shutdown("Error reading DS18B20 sensor");
-        return;
+        if (d->error_count > d->max_error_count) {
+            try_shutdown("Error reading DS18B20 sensor");
+            return;
+        } else {
+          sendf("ds18b20_result oid=%c next_clock=%u value=%i"
+                  , oid, next_begin_time, 0);
+          d->status = W1_IDLE;
```

Comment:
IMO it would make sense to go directly back to W1_READ_REQUESTED.

```suggestion
        d->request_time = request_time;
        d->status = W1_READ_REQUESTED;
```

Reply:
@functionpointer I notice you changed the line from W1_IDLE to W1_READ_REQUESTED, but didn't update the request time. I'm not sure if that was intentional or if you missed that line in my suggested change.

I was assuming that we would want to reset the 5 second timeout between error retries, but it could also make sense to time out after 5 seconds cumulatively. In fact if we used a cumulative timeout we could leave out the error count entirely.